config.lua

```diff
@@ -17,6 +17,44 @@ QBConfig.Player.ThirstRate = 3.8 -- Rate at which thirst goes down.
 QBConfig.Player.Bloodtypes = {
     "A+", "A-", "B+", "B-", "AB+", "AB-", "O+", "O-",
 }
+QBConfig.Player.IdentifierTypes = {
+    ['citizenid'] = {
+        ValueFunction = function()
+            return tostring(QBCore.Shared.RandomStr(3) .. QBCore.Shared.RandomInt(5)):upper()
+        end,
+        DatabaseColumn = 'citizenid'
+    },
+    ['AccountNumber'] = {
+        ValueFunction = function()
+            return 'US0' .. math.random(1, 9) .. 'QBCore' .. math.random(1111, 9999) .. math.random(1111, 9999) .. math.random(11, 99)
+        end,
+        DatabaseColumn = 'charinfo'
+    },
+    ['PhoneNumber'] = {
+        ValueFunction = function()
+            return math.random(100,999) .. math.random(1000000,9999999)
+        end,
+        DatabaseColumn = 'charinfo'
+    },
+    ['FingerId'] = {
+        ValueFunction = function()
+            return tostring(QBCore.Shared.RandomStr(2) .. QBCore.Shared.RandomInt(3) .. QBCore.Shared.RandomStr(1) .. QBCore.Shared.RandomInt(2) .. QBCore.Shared.RandomStr(3) .. QBCore.Shared.RandomInt(4))
+        end,
+        DatabaseColumn = 'metadata'
+    },
+    ['WalletId'] = {
+        ValueFunction = function()
+            return 'QB-' .. math.random(11111111, 99999999)
+        end,
+        DatabaseColumn = 'metadata'
+    },
+    ['SerialNumber'] = {
+        ValueFunction = function()
+            return math.random(11111111, 99999999)
+        end,
+        DatabaseColumn = 'metadata'
+    },
+}
```

Comment:
This structure seems worse to me than having the explicit functions as you did previously. Why did you make this change?

Reply:
Previously, explicit function's were only used once only, like this we can remove the remainder of the repeating code.
This also allows people to make new identifier types with their own formats by making a config option rather than creating a new explicit function and adding the sql.

Also beyond this, the config also makes it easier for people to change the formats of things like phone numbers etc without having to go digging through some code.

Reply:
Moving the shared code to a single function is a good change. The change I'm referring to is moving the generation functions from explicit functions to implied functions in this table via config. This is not getting rid of or extracting any code, but merely changing what was once an explicit function to an entry in a table. Explicit functions are better than implicit functions in a table. The biggest reason is it is easier to trace the callstack of explicit functions than to understand how the table functions are being used.

Reply:
Okay, I see your point. In my eyes, I think pushing as much values as possible to the config is never really a bad idea (as long as it remains static, which this is).

Reply:
The pattern it sets up though is an API where the user calls one function with a string argument to represent the underlying generation function. I find this layer of abstraction to be more confusing than helpful. I won't revoke my approval over it, but I stand in protest to table based functions like this.

Reply:
The table based function is simply so that I can push the formatting function into the config. There are a lot of servers out there that need to change the style of phone numbers (for example) so that they can fit the format of their host country without having to dick around with the function, but yes, this also can be done with functions similarly to before.

Again, I do see your point, and to some degree agree with it. I'd like to wait for some fge